source/OrderedTypes/PredicativeLFP.lagda

```diff
@@ -0,0 +1,1704 @@
+                         → ((Ǝ j ꞉ I₁ , Y j ⊆ X × (b , Y j) ∈ R) holds)
+  is-small-pres-forward b X = ⌜ is-small-pres b X ⌝
+
+  is-small-pres-backward : (b : B)
+                         → (X : 𝓟 {𝓥} B)
+                         → ((Ǝ j ꞉ I₁ , Y j ⊆ X × (b , Y j) ∈ R) holds)
+                         → b ≤ᴮ (⋁ 【 β , X 】)
+  is-small-pres-backward b X = ⌜ is-small-pres b X ⌝⁻¹
+
+  ϕ-is-small : (a : ⟨ L ⟩) → (b : B) → ((b , a) ∈ ϕ) is 𝓥 small
+  ϕ-is-small = pr₁ bnd
+
+  small-ϕ : (b : B) → (a : ⟨ L ⟩) → 𝓥  ̇
+  small-ϕ b a = pr₁ (ϕ-is-small a b)
+
+  small-ϕ-equiv-ϕ : (a : ⟨ L ⟩) → (b : B) → small-ϕ b a ≃ (b , a) ∈ ϕ
+  small-ϕ-equiv-ϕ a b = pr₂ (ϕ-is-small a b)
+
+  ϕ-is-small-forward : (a : ⟨ L ⟩) → (b : B) → small-ϕ b a → (b , a) ∈ ϕ
+  ϕ-is-small-forward a b = ⌜ small-ϕ-equiv-ϕ a b  ⌝
+
+  ϕ-is-small-backward : (a : ⟨ L ⟩) → (b : B) → (b , a) ∈ ϕ → small-ϕ b a
+  ϕ-is-small-backward a b = ⌜ small-ϕ-equiv-ϕ a b ⌝⁻¹
+
+  I₂ : 𝓥  ̇
+  I₂ = pr₁ (pr₂ bnd)
+
+  α : I₂ → 𝓥  ̇
+  α = pr₁ (pr₂ (pr₂ bnd))
+
+  cover-condition : ((a : ⟨ L ⟩)
+                  → (b : B)
+                  → (b , a) ∈ ϕ
+                  → (Ǝ i ꞉ I₂ , α i is-a-small-cover-of ↓ᴮ L β a) holds)
+  cover-condition = pr₂ (pr₂ (pr₂ bnd))
+
+\end{code}
+
+The following serves as evidence that the desired QIT family at least has
+sound point constructors.
+
+\begin{code}
+
+  data Small-Ind-Check : B → 𝓥  ̇ where
+   Small-c-cl' : (i : I₁)
+               → ((b : B) → (b ∈ Y i → Small-Ind-Check b))
+               → (b : B) → (b , Y i) ∈ R
+               → Small-Ind-Check b
+   Small-ϕ-cl' : (i : I₂)
+               → (m : α i → B)
+               → (b : B)
+               → small-ϕ b (⋁ (α i , β ∘ m))
+               → ((b' : B) → (b' ≤ᴮ (⋁ (α i , β ∘ m)) → Small-Ind-Check b'))
+               → Small-Ind-Check b
+
+\end{code}
+
+Again, we use records to assert the existence of another QIT family with
+apropriate induction principle.
+
+\begin{code}
+
+  record inductively-generated-small-subset-exists : 𝓤ω where
+   constructor
+    inductively-generated-small-subset
+
+   field
+    Small-Ind : B → 𝓥  ̇
+    Small-Ind-trunc : (b : B) → is-prop (Small-Ind b)
+    Small-c-cl : (i : I₁)
+               → ((b : B) → (b ∈ Y i → Small-Ind b))
+               → (b : B) → (b , Y i) ∈ R
+               → Small-Ind b
+    Small-ϕ-cl : (i : I₂)
+               → (m : α i → B)
+               → (b : B)
+               → small-ϕ b (⋁ (α i , β ∘ m))
+               → ((b' : B) → (b' ≤ᴮ (⋁ (α i , β ∘ m)) → Small-Ind b'))
+               → Small-Ind b
+    Small-Ind-Induction : (P : (b : B) → 𝓟 {𝓣} (Small-Ind b))
+                        → ((i : I₁) → (f : (x : B) → (x ∈ Y i → Small-Ind x))
+                         → ((x : B) → (y : x ∈ Y i) → f x y ∈ P x)
+                         → (b : B) → (g : (b , Y i) ∈ R)
+                         → Small-c-cl i f b g ∈ P b)
+                        → ((i : I₂)
+                         → (m : α i → B)
```

Comment:
This induction principle becomes rather difficult to understand without introducing some abbreviations.